### ISSUE TITLE
Movement Generators not reset after aura expire

### ISSUE THREAD

**Issue by ariel-** [TrinityCore/TrinityCore#13976]
The problem is the following:

When MC'd by a creature, sometimes the MoveGenerator keeps on follow, example .movegens output after being MC'd (and then killed) Harbinger Skyriss in The Arcatraz:

Player (lowguid: 1) movement generators stack:
   Idle
   Follow creature Harbinger Skyriss (lowguid 1023529)
Empty

So, even without MC aura and with follow creature dead I'm forced to walk into it.

Commit/DB:
ccdc100d10143285bc058d968f0472ea4d073120
##

**Comment by ariel-** [comment]
Try fix:

In Player::Update()
Replace
```c++
    if (IsCharmed())
        if (Unit* charmer = GetCharmer())
            if (charmer->GetTypeId() == TYPEID_UNIT && charmer->IsAlive())
                UpdateCharmedAI();
```
with
```c++
    if (IsCharmed())
    {
        if (Unit* charmer = GetCharmer())
        {
            if (charmer->GetTypeId() == TYPEID_UNIT && charmer->IsAlive())
                UpdateCharmedAI();
            else
                RemoveCharmedBy();
        }
    }
```

**Comment by ariel-** [comment]
Closed by 4e4b2b9a138650ebe6087e18dc392f08dafe4e37
